### PR TITLE
planning: use different algorithm to find current lane and current ro…

### PR DESCRIPTION
**Pull request by jmtao**
…uting index for learning data

### CODE REVIEW

File: modules/planning/pipeline/feature_generator.cc [hunk 1]
```diff
@@ -491,6 +496,11 @@ void FeatureGenerator::GenerateRoutingFeature(
   }
 
   if (routing_index < 0) {
+    std::ostringstream msg;
+    msg << "no LOCAL routing. frame_num["
+        << learning_data_frame->frame_num() << "]";
```
Comment (xiaoxq) [inline modules/planning/pipeline/feature_generator.cc:501]: Try `StrCat`.

File: modules/planning/pipeline/feature_generator.cc [hunk 2]
```diff
@@ -515,6 +525,13 @@ void FeatureGenerator::GenerateRoutingFeature(
     i++;
   }
 
+  if (local_routing_lane_ids.empty()) {
+    std::ostringstream msg;
+    msg << "no LOCAL routing. frame_num["
+        << learning_data_frame->frame_num() << "]";
```
Comment (xiaoxq) [inline modules/planning/pipeline/feature_generator.cc:531]: ditto

File: modules/planning/pipeline/feature_generator.cc [hunk 3]
```diff
@@ -264,40 +264,45 @@ void FeatureGenerator::OnRoutingResponse(
   }
 }
 
-apollo::hdmap::LaneInfoConstPtr FeatureGenerator::GetLane(
-    const apollo::common::PointENU& position,
-    int* routing_index) {
-  constexpr double kRadius = 0.1;
+int FeatureGenerator::GetADCCurrentRoutingIndex() {
+  constexpr double kRadius = 4.0;
```
Comment (xiaoxq) [inline modules/planning/pipeline/feature_generator.cc:268]: static

File: modules/planning/pipeline/feature_generator.cc (hunk 1 above)
Comment (jmtao) [reply modules/planning/pipeline/feature_generator.cc:501]: if performance is not a concern(well, it is not ,here), stream seems better here. It IS a logging output with  flexible text. 
please approve. :)

File: modules/planning/pipeline/feature_generator.cc (hunk 3 above)
Comment (jmtao) [reply modules/planning/pipeline/feature_generator.cc]: fixed. amended. thx!

**Pull request merged**